tests/testthat/helper.R

```diff
@@ -1,16 +1,37 @@
 Sys.setlocale("LC_COLLATE", "C") ## What CRAN does
 set.seed(666)
 
-## Our "test package" common harness code
-## `try` added because of some devtools::document weirdness
-try({
-    crunch_test_path <- system.file("crunch-test.R", package="crunch")
-    if (crunch_test_path == "") {
-        # hack for loadall
-        crunch_test_path <- "../../inst/crunch-test.R"
+# find a file starting with the current directory and looking (recursively) in parent folders limited to levels_up
+find_file <- function (file_name, levels_up = 3) {
+    for (i in 0:levels_up) {
+        if (i == 0) {
+            pth <- "."
+            
+        } else {
+            pth <- file.path(".", paste0(rep("..", i), collapse = "/"))
+        }
+        fls <- list.files(pth, recursive = TRUE)
+        fl <- fls[grepl(file_name, fls)]
+        if (length(fl) == 1) return(file.path(pth,fl))
+        if (length(fl)  > 1) {
+            warning("More than one file named ", file_name, " using the first one.")
+            return(file.path(pth,fl[1]))
+        }
     }
+    stop("Found no file named ", file_name, " Stopping tests because they will not function without ", file_name)
+}
+
+## Our "test package" common harness code
+crunch_test_path <- system.file("crunch-test.R", package="crunch")
+if (crunch_test_path == "") {
+    # hack for loadall
+    crunch_test_path <- find_file("crunch-test.R")
+}
+# only source crunch-test.R when the session is not interactive
+if (!interactive()) {
+    message("Loading helpers from: ", crunch_test_path)
```

Comment:
Why message? I don't want to see this every time I run tests.

Reply:
I'm happy to take it out, I was using it mostly as a confirmation that the helpers were loaded when I was making sure that the file was being sourced in all of the situations it needed to be sourced in. I thought it would be a good canary if something changes with the way that any of the different ways that tests are run that is impacting this process.